engine/table/src/test/java/io/deephaven/engine/table/impl/updateby/TestRollingSum.java

```diff
@@ -0,0 +1,1203 @@
+package io.deephaven.engine.table.impl.updateby;
+
+import io.deephaven.api.updateby.BadDataBehavior;
+import io.deephaven.api.updateby.OperationControl;
+import io.deephaven.api.updateby.UpdateByControl;
+import io.deephaven.api.updateby.UpdateByOperation;
+import io.deephaven.engine.table.ColumnDefinition;
+import io.deephaven.engine.table.PartitionedTable;
+import io.deephaven.engine.table.Table;
+import io.deephaven.engine.table.impl.*;
+import io.deephaven.engine.updategraph.UpdateGraphProcessor;
+import io.deephaven.test.types.OutOfBandTest;
+import io.deephaven.time.DateTime;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+import static io.deephaven.engine.table.impl.GenerateTableUpdates.generateAppends;
+import static io.deephaven.engine.table.impl.RefreshingTableTestCase.simulateShiftAwareStep;
+import static io.deephaven.engine.table.impl.TstUtils.*;
+import static io.deephaven.engine.util.TableTools.*;
+import static io.deephaven.function.Basic.isNull;
+import static io.deephaven.time.DateTimeUtils.convertDateTime;
+import static io.deephaven.util.QueryConstants.*;
+import static org.junit.Assert.assertArrayEquals;
+
+@Category(OutOfBandTest.class)
+public class TestRollingSum extends BaseUpdateByTest {
```

Comment:
Have you tested the coverage this gets you? It looks fairly thorough, though I didn't check it in exhaustive detail.

Reply:
100% method and line coverage for  all the `RollingSumOperator` classes when executing `TestRollingSum`.